data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/impl/IngressProducerReconcilableStore.java

```diff
@@ -113,6 +113,13 @@ public IngressProducer resolve(String host, String path) {
     return hostMapper.get(host);
   }
 
+  private String removeTrailingSlash(final String path) {
+    if (path.isEmpty() || path.charAt(path.length() - 1) != '/') {
```

Comment:
should we use the `.endsWith`? 

Or some pattern, like `replaceAll("/$", "")` ?

Just wondering...

Reply:
`endsWith` does way more than we need

Reply:
and pattern are expensive

Reply:
Migrated to use `endsWith`